sisl/grid.py

```diff
@@ -180,6 +180,68 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
         # Apply the scipy.ndimage.zoom function and return a new grid
         return self.apply(ndimage_zoom, zoom_factors, mode=mode, order=order, **kwargs)
 
+    def isosurface(self, level=None, step_size=1, **kwargs):
+        """Calculates the isosurface for a given value.
+        
+        It uses skimage.measure.marching_cubes, so you need to have scikit-image installed.
+
+        Parameters
+        ----------
+        level: float
```

Comment:
I think `level` should be a required input. The average min/max seems rather arbitrary ;)

Reply:
Yes, I thought so too, I'm going to change it.

I also thought that maybe it is useful to let the user specify the level as a percentage from min to max (0 -> min, 1 -> max) so  that the same value can be used for different grids and the user does not need to check what are the exact values contained in the grid before using the `isosurface` method. I don't know how to make it easy to accept both the value and the percentage. What do you think?

Reply:
I don't think the percentage would be of great use. For various reasons. Consider a comparison of simulations where one in the set has a much higher maximum. Then you can't compare by percentage.

Reply:
You are right, no percentage argument.